### PR TITLE
Update the webpage

### PR DESCRIPTION
**Pull request by bzsolt**
* Re-generate the files with `tools/update-webpage.sh`
 * Add the document about reference counting
 * Introduce a drop down list for documents

JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: 03.API-EXAMPLE.md [hunk 1]
```diff
@@ -521,4 +521,4 @@ Value of x is 17
 
 ## Further steps
 
-For further API description, please visit [API Reference page]({{ site.github.url }}/api-reference/) on [JerryScript home page]({{ site.github.url }}/).
+For further API description, please visit [API Reference page](https://samsung.github.io/jerryscript/api-reference/) on [JerryScript home page](https://samsung.github.io/jerryscript/).
```
Comment (LaszloLango) [inline 03.API-EXAMPLE.md]: Why was this changed?

File: 04.INTERNALS.md [hunk 2]
```diff
@@ -246,7 +246,7 @@ simple value is a pre-defined constant which can be:
 
 Compressed pointers were introduced to save heap space.
 
-![Compressed Pointer]({{ site.baseurl }}/img/ecma_compressed.png){: class="thumbnail center-block img-responsive" }
+![Compressed Pointer]({{ site.github.url }}/img/ecma_compressed.png){: class="thumbnail center-block img-responsive" }
```
Comment (LaszloLango) [inline 04.INTERNALS.md]: Are you sure is this working?

File: 03.API-EXAMPLE.md (hunk 1 above)
Comment (bzsolt) [inline 03.API-EXAMPLE.md]: In the previous commit this line has been set manually, but the webpage-generator script set this line automatically.

File: 04.INTERNALS.md (hunk 2 above)
Comment (bzsolt) [inline 04.INTERNALS.md]: Yes, I am. You can check it here: http://bzsolt.github.io/jerryscript/internals/#compressed-pointers
It is also generated by the script.

**Comment by zherczeg** [comment]
LGTM
